### DISCUSSION TITLE
How to use a theme colour within a plugin config

### DISCUSSION THREAD

**Discussion by alewis001** [LazyVim/LazyVim#4228] [Q&A] [answered]
Hi, sorry if this has been answered before; I did try looking but couldn't find anything. 

I have added a new "TODO" type in todo-comments. I'd also like to add a colour, which I have working but it is currently just a hex value for the colour code.

I'd like to refer to a colour by name such that it picks up the actual colour from the applied theme. Is there way to do that?

Here's my todo-comments config:
```lua
return {
  "folke/todo-comments.nvim",
  opts = {
    keywords = {
      ALEWIS = {
        icon = " ",
        color = "alewis",
        alt = { "[ALEWIS]" },
      },
    },
    colors = {
      alewis = { "#a6e3a1" },
    },
  },
}
```
The hex value in `alewis = { "#a6e3a1" },` is the colour "green" from the catppuccin theme. Ideally, I'd just be able to refer to "green" in some way so it's the "green" from whatever theme is active. Is that possible?

**Comment by 0x2E** [comment, marked as answer]
Extract colors from https://github.com/catppuccin/nvim#get-catppuccin-colors

<img width="697" alt="Screenshot 2024-08-05 at 10 45 20" src="https://github.com/user-attachments/assets/a6128d03-859c-499f-9bdf-f813d6659782">

**Reply by alewis001** [reply]
Thank you!

**Reply by alewis001** [reply]
Out of interest, is there a way to pick up "green" from any loaded theme or am I thinking about this in the wrong way? Would I need to pick a highlight group to use as the source of the colour instead of  specific colour like green?

**Reply by dpetka2001** [reply]
It's easier to use a well known highlight group because not all colorschemes use the same colors and you can't get the colors of the palettes they use easily, since they don't use the same paths. `GitSignsAdd` highlight group would be a good choice to use as it's almost always some kind of green in each colorscheme.

**Reply by alewis001** [reply]
That's great, thank you for the clarification.
